src/mon/mon_types.h

```diff
@@ -24,16 +24,19 @@
 #include "common/bit_str.h"
 #include "common/ceph_releases.h"
 
-#define PAXOS_MDSMAP     0
-#define PAXOS_OSDMAP     1
-#define PAXOS_LOG        2
-#define PAXOS_MONMAP     3
-#define PAXOS_AUTH       4
-#define PAXOS_MGR        5
-#define PAXOS_MGRSTAT    6
-#define PAXOS_HEALTH     7
-#define PAXOS_CONFIG     8
-#define PAXOS_NUM        9
+enum PAXOS_SVC_TYPE : int {
+  PAXOS_MDSMAP,
```

Comment:
please don't use ALL_CAPTITAL_CHARACTERS as the name of the enum type, actually it is optional. i wonder if it'd be better to just to skip it.

might want to add a comment to document that these enum are used as the indexes of `Monitor::paxos_service`.

BTW, we could probably define it as a `std::array<>` as its size is known at compile-time.

Reply:
1. I'll add comment here
2. I'll try to change the type of ```Monitor::paxos_service``` from std::vector to std::array